adafruit_rgbled.py

```diff
@@ -0,0 +1,157 @@
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`adafruit_rgbled`
+================================================================================
+
+CircuitPython driver for RGB LEDs
+
+* Author(s): Brent Rubell
+
+Implementation Notes
+--------------------
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://github.com/adafruit/circuitpython/releases
+
+* Adafruit's SimpleIO library: https://github.com/adafruit/Adafruit_CircuitPython_SimpleIO
+"""
+from pulseio import PWMOut
+from simpleio import map_range
+
+__version__ = "0.0.0-auto.0"
+__repo__ = "https://github.com/adafruit/Adafruit_CircuitPython_RGBLED.git"
+
+class RGBLED:
+    """
+    A RGB LED.
+
+    :param ~microcontroller.Pin red_pin: The red RGB LED pin to output PWM on.
+    :param ~microcontroller.Pin green_pin: The green RGB LED pin to output PWM on.
+    :param ~microcontroller.Pin blue_pin: The blue RGB LED pin to output PWM on.
+    :param bool invert_pwm: False if the RGB LED is common cathode,
+        true if the RGB LED is common anode.
+
+    Example for setting a RGB LED using a RGB Tuple (Red, Green, Blue):
+
+    .. code-block:: python
+
+        import board
+        import adafruit_rgbled
+
+        RED_LED = board.D5
+        GREEN_LED = board.D6
+        BLUE_LED = board.D7
+
+        # Create a RGB LED object
+        led = adafruit_rgbled.RGBLED(RED_LED, BLUE_LED, GREEN_LED)
+        led.color = (255, 0, 0)
+
+    Example for setting a RGB LED using a 24-bit integer (hex syntax):
+
+    .. code-block:: python
+
+        import board
+        import adafruit_rgbled
+
+        RED_LED = board.D5
+        GREEN_LED = board.D6
+        BLUE_LED = board.D7
+
+        # Create a RGB LED object
+        led = adafruit_rgbled.RGBLED(RED_LED, BLUE_LED, GREEN_LED)
+        led.color = 0x100000
+
+    Example for setting a RGB LED using a ContextManager:
+
+    .. code-block:: python
+
+        import board
+        import adafruit_rgbled
+        with adafruit_rgbled.RGBLED(board.D5, board.D6, board.D7) as rgb_led:
+            rgb_led.color = (255, 0, 0)
+
+    Example for setting a common-anode RGB LED using a ContextManager:
+
+    .. code-block:: python
+
+        import board
+        import adafruit_rgbled
+        with adafruit_rgbled.RGBLED(board.D5, board.D6, board.D7, invert_pwm=True) as rgb_led:
+            rgb_led.color = (0, 255, 0)
+
+    """
+    def __init__(self, red_pin, green_pin, blue_pin, invert_pwm=False):
+        self._rgb_led_pins = [red_pin, green_pin, blue_pin]
+        for i in range(len(self._rgb_led_pins)):
+            pin_type = str(type(self._rgb_led_pins[i]))
+            if pin_type == "<class 'Pin'>":
+                self._rgb_led_pins[i] = PWMOut(self._rgb_led_pins[i])
```

Comment:
make sure we can pass in pins OR PWMOut objects (e.g. PCA9685), test here for both types or throw an exception

Reply:
Sure, I'll add a check. Though - a PCA9685 PWM channel has the type `<class 'PWMChannel'>`, not `PWMOut`. I could catch `Pin`/`PWMChannel`/`PWMOut` objects though and raise an exception.

Reply:
maybe check there's an attribute for freq/period?

Reply:
Checked in latest commit!